### PR TITLE
Hyundai: Car Port for Ioniq Plug-in Hybrid 2019

### PR DESCRIPTION
**Pull request by sunnyhaibin**
**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/sunnyhaibin/openpilot-1/blob/180ed91b85681dcb29c257f22dc15acf51e9ebd4/selfdrive/car/tests/routes.py#L83)
- [x] route with openpilot: `3f29334d6134fcd4|2022-03-30--22-00-50`
- [x] route with stock system: `3f29334d6134fcd4|2022-03-29--19-28-39`
- [x] harness type: [Hyundai C](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide)

Thanks to community Ioniq Plug-In Hybrid owner camwow13.

### CODE REVIEW

**Comment by sshane** [comment]
Make sure you upload rlogs and preserve for all routes, there's no rlogs in the stock system route.

**Comment by adeebshihadeh** [comment]
> Make sure you always upload rlogs for all related routes, there's no rlogs in the stock system route.

Stock system doesn't matter so much. It's more of a nice to have.

**Comment by sshane** [comment]
Hmm, the only difference here is the 2019 isn't in the `use_fca` list, everything else is duplicate. I wonder if we can clean up the use_fca list and just detect by CAN fingerprint. I checked a 2020 Ioniq PHEV and it actually has both SCC12 and FCA11, so I wonder if the signals we care about (AEB/FCW) are on both and we can move `IONIQ_PHEV` out of `use_fca` and avoid adding another platform.

**Comment by sshane** [comment]
Decided we're going to merge as a new platform for now and then work on using the CAN fingerprint to clean the list up as well as merge any otherwise duplicate candidates.

File: selfdrive/car/hyundai/carcontroller.py [hunk 1]
```diff
@@ -108,7 +108,8 @@ def update(self, CC, CS):
     if self.frame % 5 == 0 and self.car_fingerprint in (CAR.SONATA, CAR.PALISADE, CAR.IONIQ, CAR.KIA_NIRO_EV, CAR.KIA_NIRO_HEV_2021,
                                                         CAR.IONIQ_EV_2020, CAR.IONIQ_PHEV, CAR.KIA_CEED, CAR.KIA_SELTOS, CAR.KONA_EV,
                                                         CAR.ELANTRA_2021, CAR.ELANTRA_HEV_2021, CAR.SONATA_HYBRID, CAR.KONA_HEV, CAR.SANTA_FE_2022,
-                                                        CAR.KIA_K5_2021, CAR.IONIQ_HEV_2022, CAR.SANTA_FE_HEV_2022, CAR.GENESIS_G70_2020, CAR.SANTA_FE_PHEV_2022):
+                                                        CAR.KIA_K5_2021, CAR.IONIQ_HEV_2022, CAR.SANTA_FE_HEV_2022, CAR.GENESIS_G70_2020,
+                                                        CAR.SANTA_FE_PHEV_2022, CAR.IONIQ_PHEV_2019):
```
Comment (sshane) [inline selfdrive/car/hyundai/carcontroller.py:112]: Are you sure this is required? The car isn't actually sending this message itself, whereas another car in this list, like the Sonata is sending this on bus 2

Comment (sunnyhaibin) [reply selfdrive/car/hyundai/carcontroller.py:112]: Good catch! You're correct, this is not required.

**Comment by sshane** [comment]
@sunnyhaibin Can you have the user test the current commit out and see how it goes?

**Comment by sunnyhaibin** [comment]
> @sunnyhaibin Can you have the user test the current commit out and see how it goes?

@sshane `3f29334d6134fcd4|2022-04-06--17-36-04` - was on stock longitudinal, rlogs uploaded!

The user tested with INDI lateral control and mentioned that the ping pong effect with the previous PIF tune was significantly decreased. He tested it on my fork with the INDI lateral tune. I've added in this car port as well:
* `3f29334d6134fcd4|2022-04-05--18-51-40`

**Comment by sshane** [comment]
Thanks for confirming the `minSteerSpeed`, I'll run final checks on the route today

**Comment by sshane** [comment]
Can you resolve the merge conflicts?

**Comment by sunnyhaibin** [comment]
@sshane Route ID with INDI changes attached!
- `3f29334d6134fcd4|2022-04-05--16-51-15`

File: selfdrive/car/hyundai/carcontroller.py [hunk 2]
```diff
@@ -109,7 +109,8 @@ def update(self, CC, CS):
     if self.frame % 5 == 0 and self.car_fingerprint in (CAR.SONATA, CAR.PALISADE, CAR.IONIQ, CAR.KIA_NIRO_EV, CAR.KIA_NIRO_HEV_2021,
                                                         CAR.IONIQ_EV_2020, CAR.IONIQ_PHEV, CAR.KIA_CEED, CAR.KIA_SELTOS, CAR.KONA_EV,
                                                         CAR.ELANTRA_2021, CAR.ELANTRA_HEV_2021, CAR.SONATA_HYBRID, CAR.KONA_HEV, CAR.SANTA_FE_2022,
-                                                        CAR.KIA_K5_2021, CAR.IONIQ_HEV_2022, CAR.SANTA_FE_HEV_2022, CAR.GENESIS_G70_2020, CAR.SANTA_FE_PHEV_2022):
+                                                        CAR.KIA_K5_2021, CAR.IONIQ_HEV_2022, CAR.SANTA_FE_HEV_2022, CAR.GENESIS_G70_2020,
+                                                        CAR.SANTA_FE_PHEV_2022):
```
Comment (sshane) [inline selfdrive/car/hyundai/carcontroller.py:113]:
```suggestion
                                                        CAR.KIA_K5_2021, CAR.IONIQ_HEV_2022, CAR.SANTA_FE_HEV_2022, CAR.GENESIS_G70_2020, CAR.SANTA_FE_PHEV_2022):
```

File: selfdrive/car/hyundai/values.py [hunk 3]
```diff
@@ -242,6 +244,24 @@ class Buttons:
...
+    ],
+    (Ecu.fwdCamera, 0x7c4, None): [
+      b'\xf1\x00AEP MFC  AT USA LHD 1.00 1.00 95740-G2400 180222',
+    ],
+    (Ecu.engine, 0x7e0, None): [
+      b'\xf1\x816H6F6051\x00\x00\x00\x00\x00\x00\x00\x00',
+    ],
+    (Ecu.transmission, 0x7e1, None): [
+      b'\xf1\x816U3J2051\x00\x00\xf1\x006U3H0_C2\x00\x006U3J2051\x00\x00PAE0G16NS1\xdbD\r\x81',
+      b'\xf1\x816U3J2051\x00\x00\xf1\x006U3H0_C2\x00\x006U3J2051\x00\x00PAE0G16NS1\x00\x00\x00\x00',
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:262]: @sunnyhaibin do you know where this came from? I can't see it in a few of his routes

Comment (sunnyhaibin) [reply selfdrive/car/hyundai/values.py:262]: @sshane Found one here! `3f29334d6134fcd4|2022-03-29--19-16-22`

**Comment by sshane** [comment]
Make sure any routes you share are on the PR's branch or stock openpilot, that INDI route was on release + your fork without the newer cereal fields we check (like latActive). But looks good other than that

**Pull request merged**